homeassistant/components/gpsd/coordinator.py

```diff
@@ -0,0 +1,64 @@
+"""Data coordinator for the GPSD integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from gps3.agps3threaded import AGPS3mechanism
+
+from homeassistant.const import (
+    ATTR_ELEVATION,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    ATTR_MODE,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import ATTR_CLIMB, ATTR_GPS_TIME, ATTR_SPEED
+
+UPDATE_INTERVAL = timedelta(minutes=1)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GpsdCoordinator(DataUpdateCoordinator):
+    """Custom coordinator for the GPSD integration."""
+
+    def __init__(self, hass: HomeAssistant, config: ConfigType) -> None:
+        """Initialize the GPSD coordinator."""
+        super().__init__(
+            hass, _LOGGER, name=config[CONF_NAME], update_interval=UPDATE_INTERVAL
+        )
+
+        host = config.get(CONF_HOST)
+        port = config.get(CONF_PORT)
+
+        self._agps_thread = AGPS3mechanism()
+        self._agps_thread.stream_data(host=host, port=port)
+        self._agps_thread.run_thread()
+
+    def _get_value(self, value: str) -> Any:
+        """Replace the GPSD response 'n/a' value with None."""
+        return None if (value is None or value == "n/a") else value
+
+    def _sync_update(self) -> dict[Any, dict[str, Any]]:
```

Comment:
In the async update we return dict[str, dict[str, Any]]. Can we change to this as well?

Reply:
Done